cpu/fe310/periph/plic.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2020 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_fe310
+ * @{
+ *
+ * @file
+ * @brief       Platform-Level interrupt controller driver
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ * @}
+ */
+
+#include "vendor/encoding.h"
+#include "vendor/platform.h"
+#include "cpu.h"
+
+/* PLIC external ISR function list */
+static plic_isr_cb_t _ext_isrs[PLIC_NUM_INTERRUPTS];
+
+static inline volatile uint32_t *_get_claim_complete_addr(void)
+{
+    uint32_t hart_id = read_csr(mhartid);
+
+    /* Construct the claim address */
+    return &PLIC_REG(PLIC_CLAIM_OFFSET +
+                     (hart_id << PLIC_CLAIM_SHIFT_PER_TARGET));
+}
+
+static inline volatile uint32_t *_get_threshold_addr(void)
+{
+    uint32_t hart_id = read_csr(mhartid);
+
+    /* Construct the claim address */
+    return &PLIC_REG(PLIC_THRESHOLD_OFFSET +
+                     (hart_id << PLIC_THRESHOLD_SHIFT_PER_TARGET));
+}
+
+void plic_enable_interrupt(unsigned irq)
+{
+    uint32_t hart_id = read_csr(mhartid);
+
+    volatile uint32_t *irq_reg = &PLIC_REG(PLIC_ENABLE_OFFSET +
+                                           (hart_id <<
+    PLIC_ENABLE_SHIFT_PER_TARGET) +
```

Comment:
indent is off here. Same in `plic_disable_interrupt`

Reply:
Blame uncrustify :disappointed: 

I reworked it into a separate register getter for the `irq_reg`, de-duplicating the register address magic in the process.